### PR TITLE
Add build event handler duration metric

### PR DESCRIPTION
**Pull request by bduffany**

### CODE REVIEW

**Review by tylerwilliams — COMMENTED** [review]
High level comments:
 - I wanted to ask about the reason for putting everything in a "metrics" package instead of having it live in a var inside the package where the metric is derived -- what does this buy us and what's the cost? It seems a little heavy to me at first glance but I'm open to it if there's a good reason!
 - Is it common to define HighVariabilityMicrosecondBuckets? They seem useful -- just checking if consumers of this data will be used to this or confused by it.

**Comment by bduffany** [comment]
@tylerwilliams 
 
> * I wanted to ask about the reason for putting everything in a "metrics" package instead of having it live in a var inside the package where the metric is derived -- what does this buy us and what's the cost? It seems a little heavy to me at first glance but I'm open to it if there's a good reason!

It buys us the ability to easily auto-generate documentation directly from the code - see https://github.com/buildbuddy-io/buildbuddy/pull/160 - the benefit of auto-generation is that it prevents the docs from going out of sync with the code and also reduces tedium of manually updating the documentation.

The cost is the possibility of this file becoming very large and compilation time of this particular file becoming a bottleneck. But this cost is only incurred during development and only for that particular file, and I figure it'll be a pretty long time before we hit that bottleneck anyway (if we ever hit it at all).

> * Is it common to define HighVariabilityMicrosecondBuckets? They seem useful -- just checking if consumers of this data will be used to this or confused by it.

The only guidance I've found (so far) for histogram buckets is to choose them based on the range of values you expect. Choose buckets too coarsely and metrics will be too low-resolution and potentially less useful. Make them too small and they will use too many resources and read/write performance may be degraded (see "Issue 2" [here](https://valyala.medium.com/improving-histogram-usability-for-prometheus-and-grafana-bc7e5df0e350))

That being said I think `HighVariabilityMicrosecondBuckets` is probably too coarse-grained and we can just use `prometheus.ExponentialBuckets` for now (done)

File: server/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	statusLabel = "status"
+)
+
+var (
+	BuildEventHandlerDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "buildbuddy",
+		Subsystem: "build_event_handler",
+		Name:      "duration",
+		Buckets:   prometheus.ExponentialBuckets(1, 10, 9),
+		Help:      "The time spent handling each build event in microseconds. Use the `_count` suffix to get the total number of build events handled.",
```
Comment (tylerwilliams) [inline server/metrics/metrics.go:21]: For this metric it's fine as-is, but for metrics that are not buildbuddy-implementation-specific, we should make sure there are easily consumable "_total" metrics. Other folks (like gitlab) do this.

**Review by tylerwilliams — APPROVED** [review]

File: server/metrics/metrics.go (hunk 1 above)
Comment (bduffany) [reply server/metrics/metrics.go:21]: Ack - will update the other PR with `_total` metrics

**Pull request merged**